### PR TITLE
Implements `move` function

### PR DESCRIPTION
**Pull request by blai**
this implements #13

### CODE REVIEW

File: test/move.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    var dirpath = path.join(__dirname, 'fixtures/nested');
+    var filepath = path.join(dirpath, filename);
+    var newdirpath = '/new/path';
+    var newfilepath = path.join(newdirpath, filename);
+    this.fs.move(dirpath, newdirpath);
+    assert.equal(this.fs.store.get(newfilepath).state, 'modified');
+    assert.throws(this.fs.read.bind(this.fs, filepath));
+  });
+
+  it('move file to an existing `to` path', function() {
```
Comment (blai) [inline test/move.js]: @SBoudrias If you disagree with this behavior, then #14 may make sense.

**Comment by SBoudrias** [comment]
Hey, that PR is on my review todo list. I'll probably come around it during the weekend.

Just to let you know, I made you a contributor on the project, so you can manage issues and review/merge other people PR.

File: actions/delete.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ module.exports = function (path, options) {
 
   var globOptions = _.extend(options.globOptions, { sync: true });
   var g = new glob.Glob(path, globOptions);
-  var files = g.found;
+  var files = g.found || [];
```
Comment (SBoudrias) [inline actions/delete.js]: Why is this needed? Also, add a unit test to confirm this is fixing a bug.

File: test/move.js (hunk 1 above)
Comment (SBoudrias) [inline test/move.js]: This seem fine, we just overwrite the `to` file correct?

File: test/move.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  });
+
+  it('throws when moving directory to an existing `to` path (as a file)', function() {
+    var filepath = path.join(__dirname, 'fixtures/file-a.txt');
+    var frompath = path.join(__dirname, 'fixtures/nested');
+
+    assert.throws(this.fs.move.bind(this.fs, frompath, filepath));
+  });
+
+  it('calls `copy` and `delete` with options', function() {
```
Comment (SBoudrias) [inline test/move.js]: We actually don't care about the implementation, only the results.

I'd remove this test.

File: actions/delete.js (hunk 2 above)
Comment (sindresorhus) [inline actions/delete.js]: This is moot. Glob always return an array.

File: test/move.js (hunk 1 above)
Comment (blai) [inline test/move.js]: Yes, but the overwrite could be unintentional, especially when it is done without giving any warning.

Comment (SBoudrias) [inline test/move.js]: Yeah, that's to be handled by write filters.
